src/test/java/com/synacy/lesson02/exercises/EnrollmentServiceTest.java

```diff
@@ -0,0 +1,104 @@
+package com.synacy.lesson02.exercises;
+
+import com.synacy.lesson02.exercises.domain.*;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class EnrollmentServiceTest {
+
+    private EnrollmentService enrollmentService;
+
+    @Mock private StudentProfileService studentProfileService;
+    @Mock private CourseClassService courseClassService;
+    @Mock private EnrollmentNotificationService enrollmentNotificationService;
+    @Mock private SystemService systemService;
+    @Mock private StudyLoadFormatter studyLoadFormatter;
+
+    @Before
+    public void setup() {
+        enrollmentService = new EnrollmentService();
+
+        enrollmentService.setStudentProfileService(studentProfileService);
+        enrollmentService.setCourseClassService(courseClassService);
+        enrollmentService.setEnrollmentNotificationService(enrollmentNotificationService);
+        enrollmentService.setSystemService(systemService);
+        enrollmentService.setStudyLoadFormatter(studyLoadFormatter);
+    }
+
+    @Test
+    public void processEnrollment_shouldUpdateStudentProfile() throws Exception {
+        EnrollmentDto enrollmentDetails = mock(EnrollmentDto.class);
+        StudentProfile studentProfile = mock(StudentProfile.class);
+
+        when(enrollmentDetails.getStudentProfile()).thenReturn(studentProfile);
+
+        enrollmentService.processEnrollment(enrollmentDetails);
+
+        verify(studentProfileService, times(1)).updateStudentProfile(
+                studentProfile
+        );
+    }
+
+    @Test
+    public void processEnrollment_shouldEnrollStudentToClass() throws Exception {
+        EnrollmentDto enrollmentDetails = mock(EnrollmentDto.class);
+
+        Student student = enrollmentDetails.getStudent();
+
+        Set<CourseClass> enrolledClasses = new HashSet<>();
+        enrolledClasses.add(mock(CourseClass.class));
+        enrolledClasses.add(mock(CourseClass.class));
+        when(enrollmentDetails.getEnrolledClasses()).thenReturn(enrolledClasses);
+        Iterator<CourseClass> courseClassIterator = enrolledClasses.iterator();
```

Comment:
A better test implementation is to get rid of the `Iterator` such as one shown below. Using the Iterator interface provides additional dependency for this test case method (this test know another class) which we would rather minimize.

![much better](https://cloud.githubusercontent.com/assets/7790345/25770776/fe4fbcbe-3270-11e7-81f6-7c27304612da.png)